card-unification.md

```diff
@@ -0,0 +1,90 @@
+# Card Unification
+
+  * [ ] Change card shape to object with `type`, `name`, `render` and optional `edit` properties
+  * [ ] Card's `type` is validated by renderer (dom renderer cannot render 'text', e.g.)
+  * [ ] Change arguments passed by editor to card's `render` (or `edit`) method
+    * [ ] single argument object with `name`, `buffer`, `isInEditor`, `options`, `editorEnv` and `payload` properties
+  * [ ] Return value of card's `render` (or `edit`) method is appended/concatenated by renderer
```

Comment:
I'm not sure we will be able to anticipate every alternative render context with this pattern, but for environments like iOS that aren't using JavaScript anyway I think it would be fine to break with this pattern and use an alternative SDK for those. I am reluctant to change the semantics of `render` to accommodate at-this-point-hypothetical alternative rendering contexts.

The thing I would like to strive for here is consistency among all JavaScript renderers — if using the return val is not the right fit then the best second option I can think of is having a buffer object that has a consistent API among all renderers (`buffer.push(stringOrDOMNode)`, or `buffer.append(stringOrDOMNode)`).